### PR TITLE
refactor: [RND-542] Convert header components styles to vanilla-extract

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
Convert different components used to compose `Header` and it's navigation to use vanilla-extract.

## Justify why they are needed
We need to use the header and navigation components in the app router.

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4047?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4047** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4047?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4046** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4046?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4045** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4045?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/Header/Header.css.ts [hunk 1]
```diff
@@ -8,3 +16,172 @@ export const focusableStyles = style({
...
+            backgroundColor: theme.colors.grayTranslucent100,
+          },
+        },
+      },
+    },
+  },
+])
+
+export const navigationSecondaryItem = style({
+  padding: `${theme.space.md} ${theme.space.md} `,
```
Comment (alebedev) [inline apps/store/src/components/Header/Header.css.ts:99]: Is is different from `padding: theme.space.md`?

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/components/Header/Header.css.ts (hunk 1 above)
Comment (Youakeem) [reply apps/store/src/components/Header/Header.css.ts:161]: Nope, I copy pasted without noticing. Will update

**Pull request merged**